### DISCUSSION TITLE
go get broken?

### DISCUSSION THREAD

**Discussion by gurgeous** [mat/besticon#58] [Q&A]
BTW, I tried a fresh go get and it came back with this:

```
$ go get -u github.com/mat/besticon/...
# github.com/mat/besticon/besticon
go/src/github.com/mat/besticon/besticon/caching.go:28:22: cannot use siteURL (type string) as type context.Context in argument to iconCache.Get:
        string does not implement context.Context (missing Deadline method)
go/src/github.com/mat/besticon/besticon/caching.go:46:24: undefined: groupcache.Context
go/src/github.com/mat/besticon/besticon/caching.go:74:79: cannot convert generatorFunc (type func(<T>, string, groupcache.Sink) error) to type groupcache.GetterFunc
```

Once again I'm sorta limited by my golang skills... did I break something? Is this a dependency issue? I can dig into this and try to fix if you want.

**Comment by mat** [comment]
Good catch, this was broken indeed... Fixed in 174b75e.

**Reply by gurgeous** [reply]
So it was a dependency issue? Could I add a test for this somehow? On my machine I basically did this:

```sh
$ mv go go.old
$ go get ...
```

**Reply by abhishekjaiswalpepper** [reply]
Is is resolved?
Still getting the same error
I have installed using `go get github.com/mat/besticon`
`# github.com/mat/besticon/besticon
../../go/pkg/mod/github.com/mat/besticon@v3.12.0+incompatible/besticon/caching.go:28:23: cannot use siteURL (variable of type string) as type context.Context in argument to iconCache.Get:
	string does not implement context.Context (missing Deadline method)
../../go/pkg/mod/github.com/mat/besticon@v3.12.0+incompatible/besticon/caching.go:47:13: impossible type assertion: ctx.(string)
	string does not implement context.Context (missing Deadline method)`

go.mod. Mod File for current module
```
module example.com

go 1.19

require github.com/mat/besticon v3.12.0+incompatible

require (
	github.com/PuerkitoBio/goquery v1.8.1 // indirect
	github.com/andybalholm/cascadia v1.3.1 // indirect
	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
	github.com/golang/protobuf v1.5.3 // indirect
	golang.org/x/image v0.7.0 // indirect
	golang.org/x/net v0.9.0 // indirect
	golang.org/x/text v0.9.0 // indirect
	google.golang.org/protobuf v1.26.0 // indirect
)



```

**Reply by abhishekjaiswalpepper** [reply]
go get github.com/mat/besticon is installing v3.12.0+incompatible

go get github.com/mat/besticon@lastest is not working getting 
` go: github.com/mat/besticon@lastest: invalid version: unknown revision lastest`

go get github.com/mat/besticon@master working properly

Any reasons for it?

**Reply by mat** [reply]
#96 fixed this: `go get github.com/mat/besticon/v3/...`
